amy/fiscal/tests/test_membership.py

```diff
@@ -343,10 +407,57 @@ def test_new_membership_has_no_members(self):
 
         latest_membership = Membership.objects.order_by("-id").first()
         self.assertEqual(response.context["membership"], latest_membership)
-        self.assertEqual(latest_membership.member_set.count(), 0)
+        self.assertEqual(latest_membership.member_set.count(), 1)
+        member = latest_membership.member_set.first()
+        self.assertEqual(member.role.name, "contract_signatory")
+        self.assertEqual(member.organization, self.org_alpha)
 
-    def test_adding_new_members(self):
-        self.setUpMembership()
+    def test_adding_new_member_to_nonconsortium(self):
+        self.setUpMembership(consortium=False)
+        self.assertEqual(self.membership.member_set.count(), 0)
+        # only 1 member allowed
+        data = {
+            "form-TOTAL_FORMS": 1,
+            "form-INITIAL_FORMS": 0,
+            "form-MIN_NUM_FORMS": 0,
+            "form-MAX_NUM_FORMS": 1000,
+            "form-0-organization": self.org_alpha.pk,
+            "form-0-role": self.member_role.pk,
+            "form-0-id": "",
+        }
+        response = self.client.post(
+            reverse("membership_members", args=[self.membership.pk]),
+            data=data,
+            follow=True,
+        )
+
+        self.assertRedirects(
+            response, reverse("membership_details", args=[self.membership.pk])
+        )
+        self.assertEqual(self.membership.member_set.count(), 1)
+        self.assertEqual(list(self.membership.organizations.all()), [self.org_alpha])
+
+        # posting this will fail
```

Comment:
Yes, this is confusing because we're not using correct HTTP statuses here.

If you post a form which contains some errors, the form won't validate and therefore it will return "200 OK" with the form and some error messages.

If the form validates, it will redirect to somewhere else.